src/app/campaign/store/actions/campaign-action.service.ts

```diff
@@ -93,6 +93,8 @@ export class CampaignActionService implements OnDestroy {
         (a === undefined || a === null || a === '' || a.length === 0) &&
         (b === undefined || b === null || b === '' || b.length === 0)
       );
+    } else if (isMoment(a) && isMoment(b)) {
```

Comment:
comparing the endAt values as Object was always giving false positive for hasChanged

Reply:
Nice! I had noticed that I was always getting an extra change cycle after the first change of any field due to the moment objects differing.

Reply:
yeah, saw a couple things once I stepped thru it a little bit and that was one of em, comparing the whole Moment and being off 😬